src/main/java/com/musala/sdcs/jdbc/DeviceRepository.java

```diff
@@ -0,0 +1,210 @@
+package com.musala.sdcs.jdbc;
+
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.musala.sdcs.device.Device;
+import com.musala.sdcs.device.builder.DeviceBuilderImpl;
+import com.musala.sdcs.device.channel.base.Channel;
+import com.musala.sdcs.device.channel.base.ChannelCreator;
+
+/**
+ * This class takes care of DB queries related to devices
+ *
+ */
+public class DeviceRepository {
+	private static final String QUERY_GET_DEVICES = "SELECT * FROM devices;";
+	private static final String QUERY_GET_DEVICES_CHANNELS = "SELECT * FROM devices_channels WHERE deviceId = %s;";
+	private static final String QUERY_GET_CHANNELS = "SELECT * FROM channels WHERE id = %s;";
+	private static final String QUERY_GET_CHANNEL_TYPE = "SELECT * FROM channel_types WHERE id = %s";
+	private static final String QUERY_GET_MODEL = "SELECT * FROM models WHERE id = %s;";
+	private static final String QUERY_GET_MANUFACTURER = "SELECT * FROM manufacturers WHERE id = %s;";
+	
+	private Connection dbConnection;
+
+	public DeviceRepository() {
+		this.dbConnection = DbUtil.getConnection();
+	}
+
+	public List<Device> getAllDevices() {
+		List<Device> devices = new ArrayList<Device>();
+		ResultSet r = executeQuery(QUERY_GET_DEVICES);
+		
+		try {
+			while (r.next()) {
+				Integer deviceId = r.getInt("id");
```

Comment:
I don't think extracting will be possible and needed. Since i take properties like model and manufacturer from another methods, I will have to call those metods from the extracted device creator, and this obscures single responsibility also. Device creaton logic is extracted in the DeviceBuilder class and facading the call to DeviceBuilder into another method won't do any good imo.